TLM/TLM/LoadingExtension.cs

```diff
@@ -333,21 +347,37 @@ public class ManualHarmonyPatch {
                 Flags.OnLevelUnloading();
                 GlobalConfig.OnLevelUnloading();
 
+                var gameObject = UIView.GetAView().gameObject;
+                void Destroy<T>() where T : MonoBehaviour {
+                    Object obj = (Object)gameObject.GetComponent<T>();
+                    if (obj != null) {
+                        Object.Destroy(obj);
+                    }
+                }
+
                 // remove vehicle button
-                var removeVehicleButtonExtender = UIView
-                                                  .GetAView().gameObject
-                                                  .GetComponent<RemoveVehicleButtonExtender>();
-                if (removeVehicleButtonExtender != null) {
-                    Object.Destroy(removeVehicleButtonExtender, 10f);
+                Destroy<RemoveVehicleButtonExtender>();
+                Destroy<RemoveCitizenInstanceButtonExtender>();
+
+                //simManager?.Remove(CustomPathManager);
+
+                //Object.Destroy(CustomPathManager);
+                //CustomPathManager = null;
+
+                Log.Info("Removing Controls from UI.");
+                if (BaseUI != null) {
+                    BaseUI.Close();
+                    Object.Destroy(BaseUI);
+                    Log._Debug("removed UIBase instance.");
+                    BaseUI = null;
                 }
 
-                // remove citizen instance button
-                var removeCitizenInstanceButtonExtender = UIView
-                                                          .GetAView().gameObject
-                                                          .GetComponent<RemoveCitizenInstanceButtonExtender>();
-                if (removeCitizenInstanceButtonExtender != null) {
-                    Object.Destroy(removeCitizenInstanceButtonExtender, 10f);
+                if (TransportDemandUI != null) {
+                    UIView uiView = UIView.GetAView();
+                    Object.Destroy(TransportDemandUI);
```

Comment:
Same risk of mem leak here; check if any event handlers are cleared before destroying.